mattermost.Dockerfile

```diff
@@ -134,9 +134,20 @@ RUN if [ "$image_flavour" = canonical ]; then \
 
 RUN rm -rf /canonical_flavour_tmp
 
+# Enable local mode
+ARG local_mode=false
+RUN if [ "$local_mode" = "true" ]; then \
+      /usr/bin/sed -i \
+        -e 's|"EnableLocalMode": false,|"EnableLocalMode": true,|g' \
+        -e 's|"LocalModeSocketLocation": "/var/tmp/mattermost_local.socket"|"LocalModeSocketLocation": "/mattermost/run/local.socket"|g' \
+        /mattermost/config/config.json ; \
+      mkdir /mattermost/run ; \
+      chown mattermost: /mattermost/run ; \
+    fi
+
```

Comment:
I think the local mode can also be enabled by the `MM_SERVICESETTINGS_ENABLELOCALMODE` environment variable and the local mode socket location would be `/var/tmp/mattermost_local.socket` by default.

One possible way to configure the local mode is to add another config, for example `extra_env` that accept a json or dotenv input and the content `extra_env` will be merged into the mattermost env. With this configuration, you can also drop the `s3_tls` config.

Reply:
I tried adding the `MM_SERVICESETTINGS_ENABLELOCALMODE` environment variables and I can see the `/var/tmp/mattermost_local.socket` created by the mattermost? Do you mean that `LocalModeSocketLocation` does not change the socket location and you want the socket to be created in the location other than `/var/tmp/mattermost_local.socket`?